components/DBTableMocker.php

```diff
@@ -27,28 +34,36 @@ public function getTableDescription(string $tableName):array {
 
             $query      = new Query("DESCRIBE `$tableName`");
 
-            $result  = $connection->query($query->getQueryString());
-            $columns = $result->fetch_all(MYSQLI_ASSOC);
+            $result        = $connection->query($query->getQueryString());
+            $columns       = $result->fetch_all(MYSQLI_ASSOC);
+            
+            foreach ($columns as $column) {
+                $this->columns[$column["Field"]] = $column;
+            }
 
             return $columns;
         }
 
         /**
-         * Sets the current column
+         * Sets the column which is being modified
          *
-         * @param string $column
-         * @return void
+         * @param string $columnName
+         * @return self
          */
-        public function column(string $column):void {
-            $this->currentColumn = $column;
+        public function column(string $columnName):self {
+            if (array_key_exists($columnName, $this->columns)) {
+                $this->currentColumn = $this->columns[$columnName];
+            }
+
+            return $this;
         }
 
         /**
-         * Returns the current column
+         * Returns the current column's name
          *
          * @return string
          */
         public function getCurrentColumn():string {
-            return $this->currentColumn;
+            return $this->currentColumn["Field"] ?? "";
```

Comment:
Due to the fact that the current column is an array now the return from this method must be a string(name of the column). If the column has not been configured yet then an empty string is returned. This helps to encapsulate the implementation of the current column.